engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlanConfiguration.java

```diff
@@ -2435,7 +2435,7 @@ private void setAttributes(List<? extends Conf> confs, Attribute.AttributeMap at
           else {
             if (processedAttributes.add(attribute)) {
               String val = e.getValue();
-              if (val.trim().charAt(0) == '{') {
+              if (val.trim().charAt(0) == '{' && !(attribute.codec instanceof JsonStringCodec)) {
```

Comment:
Nested objects parsing was failing with default JSON parsing. It needed className for correct parsing, so added JsonStringCodec.